### PR TITLE
unix: protect fork in uv_spawn from signals

### PR DESCRIPTION
**Pull request by vtjnash**
Years ago, we found that various kernels (linux, macOS) were known to
fail if they try to deliver a signal during this syscall, so we prevent
that from happening. They may have fixed those issues, but it is
generally just a bad time for signals to arrive (glibc blocks them here,
in posix_spawn for example, and even includes some more internal ones
that it won't let us touch here--but it is also using vfork, which has
additional constraints and requirements).

Refs: https://github.com/timholy/IProfile.jl/issues/16

This was initially https://github.com/JuliaAttic/libuv-archive/commit/c45ec4715ecaf6f5ae01e8a760db54283e9b27d5, but has been modified a few times since (particularly here).

### CODE REVIEW

File: src/unix/process.c [hunk 1]
```diff
@@ -326,10 +326,8 @@ static void uv__process_child_init(const uv_process_options_t* options,
 
   /* Reset signal mask. */
   sigemptyset(&set);
-  err = pthread_sigmask(SIG_SETMASK, &set, NULL);
-
-  if (err != 0)
-    uv__write_errno(error_fd);
+  if (sigprocmask(SIG_SETMASK, &set, NULL) != 0)
```
Comment (bnoordhuis) [inline src/unix/process.c:329]: Is there a reason for using `sigprocmask()` instead of `pthread_sigmask()`?

File: src/unix/process.c [hunk 2]
```diff
@@ -359,6 +357,8 @@ int uv_spawn(uv_loop_t* loop,
   int exec_errorno;
   int i;
   int status;
+  sigset_t sigiset;
+  sigset_t sigset;
```
Comment (bnoordhuis) [inline src/unix/process.c:361]: Consider renaming this. `sigset` is an obsolete-ish alternative to `sigaction()` but I think POSIX allows it to be implemented as a macro (which would expand to nonsense here.)

**Review by bnoordhuis — APPROVED** [review]

File: src/unix/process.c (hunk 1 above)
Comment (vtjnash) [reply src/unix/process.c:329]: pthreads isn't usually marked async-safe (though this function is, as this particular function is defined to just be a trivial wrapper around `sigprocmask`) and so may not be accessible after the fork, so sigprocmask seemed formally more correct here

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/2190/

**Comment by vtjnash** [comment]
I have changed this to be somewhat more conservative and avoids blocking fatal mistakes (such as SEGV).

Note that this seems to be required to make the faster code in https://github.com/libuv/libuv/pull/3257 not hit a kernel bug quite as often, so I plan to merge this as soon as possible.

CI: https://ci.nodejs.org/job/libuv-test-commit/2192/

**Review by bnoordhuis — APPROVED** [review]

**Pull request merged**